openshift/openshift.go

```diff
@@ -396,17 +396,17 @@ func (d *openshiftImageDestination) ReapplyBlob(info types.BlobInfo) (types.Blob
 // FIXME? This should also receive a MIME type if known, to differentiate between schema versions.
 // If the destination is in principle available, refuses this manifest type (e.g. it does not recognize the schema),
 // but may accept a different manifest type, the returned error must be an ManifestTypeRejectedError.
-func (d *openshiftImageDestination) PutManifest(m []byte) error {
+func (d *openshiftImageDestination) PutManifest(m []byte, instanceDigest *digest.Digest) error {
 	manifestDigest, err := manifest.Digest(m)
 	if err != nil {
 		return err
 	}
 	d.imageStreamImageName = manifestDigest.String()
 
-	return d.docker.PutManifest(m)
+	return d.docker.PutManifest(m, instanceDigest)
 }
 
-func (d *openshiftImageDestination) PutSignatures(signatures [][]byte) error {
+func (d *openshiftImageDestination) PutSignatures(signatures [][]byte, instanceDigest *digest.Digest) error {
 	if d.imageStreamImageName == "" {
```

Comment:
Corresponding to `openshiftImageSource.GetSignatures`, this should use `*instanceDigest` if available and `d.imageStreamImageName` if not.

(Well, I guess. I’m not quite sure anyone has tested multi-arch images with OpenShift to be honest.)

Reply:
Right, makes sense.  I'm not sure of that, either.